src/test/functional/features/admin-court-addresses.feature

```diff
@@ -0,0 +1,97 @@
+    Then I enter the secondary court address "<secondary address>" in the Address textbox
+    Then I enter the secondary address town "<secondary town>"
+    Then I enter the secondary address postcode "<secondary postcode>"
+    And I click the Save Addresses button
+    Then A green message is displayed for the updated address "Addresses updated"
+
+    Examples:
+      | address                       | welsh address   | town     | welsh town   | postcode | secondary address             |secondary town | secondary postcode |
+      | The Court House, Cwmbach Road | stdtduguguhguhu | Aberdare | fctgfjyfyjgv | CF44 0JE | The Court House, Cwmbach Road | Aberdare      | CF44 0JE           |
+
+  Scenario Outline: Adding secondary address and primary address both of type: Visit us
+    Given I will make sure to clear all entries of the primary address
+    When I select the Address Type "5880"
+    Then I enter court "<address>" in the Address textbox
+    Then I enter "<welsh address>" in the Address Welsh textbox
+    Then I enter "<town>" in the Town textbox
+    Then I enter "<welsh town>" in the town Welsh textbox
+    Then I enter "<postcode>" in the postcode textbox
+    When I will make sure to clear all entries for secondary address
+    Then I select the secondary address type as "5880"
+    Then I enter the secondary court address "<secondary address>" in the Address textbox
+    Then I enter the secondary address town "<secondary town>"
+    Then I enter the secondary address postcode "<secondary postcode>"
+    And I click the Save Addresses button
+    Then The error message display is "Only one visit address is permitted."
+
+    Examples:
+      | address                       | welsh address   | town     | welsh town   | postcode | secondary address             |secondary town | secondary postcode |
+      | The Court House, Cwmbach Road | stdtduguguhguhu | Aberdare | fctgfjyfyjgv | CF44 0JE | The Court House, Cwmbach Road | Aberdare      | CF44 0JE           |
+
+
+  Scenario Outline: Adding incomplete addresses (leaving primary address, secondary town and secondary postcode blank)
+    Given I will make sure to clear all entries of the primary address
+    When I will make sure to clear all entries for secondary address
+    When I select the Address Type "5880"
+    Then I enter "<town>" in the Town textbox
+    Then I enter "<postcode>" in the postcode textbox
+    Then I select the secondary address type as "5881"
+    Then I enter the secondary court address "<secondary address>" in the Address textbox
+    And I click the Save Addresses button
+    Then The error message display is "Primary Address: Address is required." "Secondary Address: Town is required." "Secondary Address: Postcode is required."
+
+    Examples:
+      | town     |  postcode | secondary address             |
+      | Aberdare |  CF44 0JE | The Court House, Cwmbach Road |
+
+  Scenario Outline: Adding primary address with invalid postcode
```

Comment:
Is it worth us testing the secondary postcode to check that we get an error message if it is invalid?

Reply:
To cover one unhappy path I have added the test for the primary address, as the scenario is similar for the secondary address so I think there is no need to add a test for the secondary address.